lib/common/socket/evloop.c.h

```diff
@@ -553,8 +553,7 @@ static void run_socket(struct st_h2o_evloop_socket_t *sock)
             socklen_t l = sizeof(so_err);
             so_err = 0;
             if (getsockopt(sock->fd, SOL_SOCKET, SO_ERROR, &so_err, &l) != 0 || so_err != 0) {
-                /* FIXME lookup the error table */
-                err = h2o_socket_error_conn_fail;
+                err = strerror(so_err);
```

Comment:
Due to the following reasons, I'm not sure if we can do this:
* We cannot assume that the string returned by `strerror` would never be overwritten. That's why we have `strerror_r`.
* We use string pointers (like `h2o_socket_error_conn_fail`) to indicate particular errors, and compare the address of the string to determine the type of the error. Use of `strerror` breaks that approach.

I'm not necessarily opposed to having more specific errors, but they should be defined as `const char []` like `h2o_socket_error_conn_fail, and we should select particular constant string based on the value of `so_err`.

Reply:
I can switch to `strerror_r` but that wouldn't help for the second point; sounds like this PR should be abandoned.

Reply:
Yeah using `strerror_r` would not solve the problem either because current API does not provide indication regarding when the buffer containing the error message can be freed.

Anyways closing the issue.